go/service/chat_local.go

```diff
@@ -386,223 +357,128 @@ func (h *chatLocalHandler) GetInboxSummaryLocal(ctx context.Context, arg chat1.G
-			return
-		}
-		conversations = append(conversations, info)
-	}
-
-	query := &chat1.GetInboxQuery{}
+func (h *chatLocalHandler) localizeConversation(
+	ctx context.Context, conversationRemote chat1.Conversation) (
+	conversationLocal chat1.ConversationLocal, err error) {
 
-	if len(criteria.TlfName) != 0 {
-		// TODO: do some caching in boxer so we don't end up calling this RPC
-		// unnecessarily too often
-		resp, err := h.boxer.tlf.CryptKeys(ctx, criteria.TlfName)
-		if err != nil {
-			return nil, err
-		}
-		tlfIDb := resp.TlfID.ToBytes()
-		if tlfIDb == nil {
-			return nil, errors.New("invalid TLF ID acquired")
-		}
-		criteria.TlfName = string(resp.CanonicalName)
-
-		tlfID := chat1.TLFID(tlfIDb)
-		query.TlfID = &tlfID
-	}
-
-	if criteria.Visibility != chat1.TLFVisibility_ANY {
-		query.TlfVisibility = &criteria.Visibility
-	}
-
-	conversationsRemote, err := h.remoteClient().GetInboxRemote(ctx, chat1.GetInboxRemoteArg{
-		Query:      query,
-		Pagination: nil,
-	})
-	*rlimits = append(*rlimits, conversationsRemote.RateLimit)
-	if err != nil {
-		return nil, err
-	}
-	convsLocal, err := h.resolveConversationsPipeline(ctx, conversationsRemote.Inbox.Conversations)
-	if err != nil {
-		return nil, err
-	}
-	for _, convLocal := range convsLocal {
-		if len(criteria.TlfName) > 0 && convLocal.Info.TlfName != criteria.TlfName {
-			// check again using signed information to make sure it's the correct
-			// conversation
-			return nil, libkb.UnexpectedChatDataFromServer{Msg: fmt.Sprintf("Unexpected data is returned from server. We asked for %v, but got conversation for %v. TODO: handle tlfName changes properly for SBS case", criteria.TlfName, convLocal.Info.TlfName)}
-		}
-		appendMaybe(*convLocal.Info)
+	conversationLocal.Info = chat1.ConversationInfoLocal{
+		Id: conversationRemote.Metadata.ConversationID,
 	}
 
-	h.G().Log.Debug("- resolveConversations: returning: %d messages", len(conversations))
-	return conversations, nil
-}
-
-func (h *chatLocalHandler) getConversationInfoByID(ctx context.Context, id chat1.ConversationID,
-	rlimits *[]*chat1.RateLimit) (conversationInfo chat1.ConversationInfoLocal, maxMessages []chat1.MessageFromServerOrError, err error) {
-	res, err := h.remoteClient().GetInboxRemote(ctx, chat1.GetInboxRemoteArg{
-		Query: &chat1.GetInboxQuery{
-			ConvID: &id,
-		},
-	})
-	*rlimits = append(*rlimits, res.RateLimit)
-	if err != nil {
-		return chat1.ConversationInfoLocal{}, nil, err
+	if len(conversationRemote.MaxMsgs) == 0 {
+		errMsg := "conversation has an empty MaxMsgs field"
+		return chat1.ConversationLocal{Error: &errMsg}, nil
 	}
-	if len(res.Inbox.Conversations) == 0 {
-		return chat1.ConversationInfoLocal{}, nil, fmt.Errorf("unknown conversation: %v", id)
+	if conversationLocal.MaxMessages, err = h.unboxMessages(ctx, conversationRemote.MaxMsgs); err != nil {
+		errMsg := err.Error()
+		return chat1.ConversationLocal{Error: &errMsg}, nil
```

Comment:
So this still prints out errors for each line in the CLI output like my last PR did?